Updated documentation related to apt-cacher-ng

Akasurde opened the pull request:
Fixes #11556 

Signed-off-by: Abhijeet Kasurde akasurde@redhat.com

mauri committed:
Fix files ownership when ADD is used

Signed-off-by: Mauricio Garavaglia <mauriciogaravaglia@gmail.com>

mauri committed:
Fix files ownership when ADD is used

Signed-off-by: Mauricio Garavaglia <mauriciogaravaglia@gmail.com>

Commit:
Improvement for docker subcommand's help messages

Signed-off-by: Wen Cheng Ma <wenchma@cn.ibm.com>

Commit:
fixed links in Digital Ocean example

Signed-off-by: Victoria Bialas <victoria.bialas@docker.com>

jecarey committed:
Use of checkers in docker_cli_pull_test.go

Signed-off-by: James Carey <jecarey@us.ibm.com>

dnephin committed:
Remove the graph driver from the daemon, move it into the layer store.

Support restoreCustomImage for windows with a new interface to extract
the graph driver from the LayerStore.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

Govinda-Fichtner committed:
update Dockerfile to reflect latest changes

Signed-off-by: Govinda Fichtner <govinda.fichtner@googlemail.com>

StefanScherer committed:
Move DOCKER_ENGINE_ envs to integration-daemon-setup to allow build w/o Makefile

Signed-off-by: Stefan Scherer <scherer_stefan@icloud.com>

donovan committed:
fix incorrect ip addresses in networkingcontainers

Signed-off-by: Donovan Jones <git@gamma.net.nz>

dnephin committed:
Move graph driver registration out of the daemon package.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

Commit:
Fix the restore data example
Fixes: #18872

For backup, restore case, we usually restore in same directory, so
when we untar it, we not need go to /dbdata directory, wich could
result in /dbdata/dbdata, this fix make it untar directly. The directory
/dbdata would be same as old container /dbdata.

Signed-off-by: Kai Qiang Wu(Kennan) <wkqwu@cn.ibm.com>

thaJeztah committed:
Be more explicit about "+1" comments

Add some more information about not leaving random
"+1" comments.

Also removed the requirements to send "uname -a",
because that information is now included in the output
of "docker version".

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Commit:
Update integration tests when container and image have same name

Signed-off-by: Wen Cheng Ma <wenchma@cn.ibm.com>

jessfraz committed:
fix default profile where unsupported

Signed-off-by: Jessica Frazelle <acidburn@docker.com>

jessfraz committed:
fix libseccomp where version < 2.2.1

Signed-off-by: Jessica Frazelle <acidburn@docker.com>

zhanghuanzhong committed:
print the registry name.

Signed-off-by: Huanzhong Zhang <zhanghuanzhong90@gmail.com>

coolljt0725 committed:
Add DOCKER_BUILD_ARGS env to pass build-arg for building deb and rpm

Signed-off-by: Lei Jitang <leijitang@huawei.com>

justincormack committed:
Allow use of robust list syscalls

The set_robust_list syscall sets the list of futexes which are
cleaned up on thread exit, and are needed to avoid mutexes
being held forever on thread exit.

See for example in Musl libc mutex handling:
http://git.musl-libc.org/cgit/musl/tree/src/thread/pthread_mutex_trylock.c#n22

Signed-off-by: Justin Cormack <justin.cormack@unikernel.com>

Akasurde commented:
- Added description of dockerhost variable
- Added example with --link flag

Akasurde committed:
Updated documentation related to apt-cacher-ng

- Added description of dockerhost variable
- Added example with --link flag

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Updated documentation related to apt-cacher-ng

- Added description of dockerhost variable
- Added example with --link flag

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

vdemeester commented:
ping @thaJeztah

justincormack committed:
Block original umount syscall in default seccomp filter

The original umount syscall without flags argument needs to
be blocked too.

Signed-off-by: Justin Cormack <justin.cormack@unikernel.com>

justincormack committed:
Deny finit_module in default seccomp profile

This is a new version of init_module that takes a file descriptor
rather than a file name.

Signed-off-by: Justin Cormack <justin.cormack@unikernel.com>

justincormack committed:
Block clock_adjtime in default seccomp config

clock_adjtime is the new posix style version of adjtime allowing
a specific clock to be specified. Time is not namespaced, so do
not allow.

Signed-off-by: Justin Cormack <justin.cormack@unikernel.com>

Commit:
Fix error messages

`docker kill 123` will show something like:
`Error response from daemon: Cannot kill container 123: nosuchcontainer: No such container: 123`
Notice the `nosuchcontainer` text, that should not be there as that's an internal ID that means nothing to the end user.
This PR fixes this by using `util.GetErrorMessage()` to extract just the message.

While in that dir I found a couple of other spots that could use the same call, just to be safe.

Signed-off-by: Doug Davis <dug@us.ibm.com>

justincormack committed:
Block stime in default seccomp profile

The stime syscall is a legacy syscall on some architectures
to set the clock, should be blocked as time is not namespaced.

Signed-off-by: Justin Cormack <justin.cormack@unikernel.com>

justincormack committed:
Block vm86 syscalls in default seccomp profile

These provide an in kernel virtual machine for x86 real mode on x86
used by one very early DOS emulator. Not required for any normal use.

Signed-off-by: Justin Cormack <justin.cormack@unikernel.com>

jessfraz committed:
add note to packagers.md about libseccomp version

Signed-off-by: Jessica Frazelle <acidburn@docker.com>

justincormack committed:
Block additional ptrace related syscalls in default seccomp profile

Block kcmp, procees_vm_readv, process_vm_writev.
All these require CAP_PTRACE, and are only used for ptrace related
actions, so are not useful as we block ptrace.

Signed-off-by: Justin Cormack <justin.cormack@unikernel.com>

tianon committed:
Merge pull request #18728 from hypriot/update-dockerfile-for-arm

update Dockerfile.arm to reflect latest changes in main Dockerfile

justincormack committed:
Block bpf syscall from default seccomp profile

The bpf syscall can load code into the kernel which may
persist beyond container lifecycle. Requires CAP_SYS_ADMIN
already.

Signed-off-by: Justin Cormack <justin.cormack@unikernel.com>

Commit:
Merge pull request #18968 from justincormack/stime

Block stime in default seccomp profile

Commit:
Merge pull request #18961 from justincormack/clock_adjtime

Block clock_adjtime in default seccomp config

Commit:
Merge pull request #18959 from justincormack/finit_module

Deny finit_module in default seccomp profile

calavera committed:
Merge pull request #18950 from zhanghuanzhong/docker-fix-pull-image-log-info

Print the registry name while pulling an image

Commit:
Merge pull request #18956 from justincormack/umount

Block original umount syscall in default seccomp filter

Commit:
Merge pull request #18953 from justincormack/robust_list

Allow use of robust list syscalls in default seccomp policy

Commit:
Merge pull request #18947 from jfrazelle/fix-seccomp-unsupported

fix default profile where unsupported

Commit:
Merge pull request #18949 from jfrazelle/fix-libseccomp-version

fix libseccomp where version < 2.2.1

jessfraz committed:
remove seccomp from seccomp profile

This can be allowed because it should only restrict more per the seccomp docs, and multiple apps use it today.

Signed-off-by: Jessica Frazelle <acidburn@docker.com>

Commit:
Merge pull request #18933 from coolljt0725/always_build

Add DOCKER_BUILD_ARGS env to pass build-arg for building deb and rpm

Commit:
Merge pull request #18971 from justincormack/ptrace

Block additional ptrace related syscalls in default seccomp profile

Commit:
Merge pull request #18972 from justincormack/bpf

Block bpf syscall from default seccomp profile

Commit:
Merge pull request #18969 from justincormack/vm86

Block vm86 syscalls in default seccomp profile

Commit:
Merge pull request #18877 from dnephin/move_graph_driver_to_layer_store

Move graph driver to layer store

vdemeester committed:
Add authorization plugins to docker info

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

Commit:
Merge pull request #18974 from jfrazelle/remove-seccomp-from-seccomp-profile

remove seccomp from seccomp profile

tswift242 committed:
Fix typo in daemon storage-driver docs

Signed-off-by: Thomas Swift <tgs242@gmail.com>

calavera committed:
Merge pull request #18965 from duglin/FixError

Fix error messages

thaJeztah committed:
Merge pull request #18978 from tswift242/fix-daemon-storage-doc-typo

Fix typo in daemon storage-driver docs

Commit:
Merge pull request #18627 from londoncalling/cloud-installs

[WIP] docs updates per issue 18282 cloud installs

calavera committed:
Remove usage of pkg sockets and tlsconfig.

- Use the ones provided by docker/go-connections, they are a drop in replacement.
- Remove pkg/sockets from docker.
- Keep pkg/tlsconfig because libnetwork still needs it and there is a
  circular dependency issue.

Signed-off-by: David Calavera <david.calavera@gmail.com>

hqhq committed:
Fix docs for memory-swap

Fixes: #18894

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

coolljt0725 committed:
Fix docker stats show wrong memory limit when do docker update

When a container create with -m 100m and then docker update other
cgroup settings such as --cpu-quota, the memory limit show by
docker stats will become the default value but not the 100m.

Signed-off-by: Lei Jitang <leijitang@huawei.com>

cpuguy83 committed:
Merge pull request #18981 from calavera/stop_using_sockets_package

Remove pkg sockets and tlsconfig.

coolljt0725 committed:
Fix daemon failed to start with error "layer does not exist"

Signed-off-by: Lei Jitang <leijitang@huawei.com>

hqhq committed:
Handle error for dockerCmdInDir

Only two of these are not handled, the one in `TestBuildForceRm`
is intended to not be handled, while the other one in
`TestBuildResourceConstraintsAreUsed` causes problem.

In test case `TestBuildResourceConstraintsAreUsed`, somehow we
are not able to access network to get base image, but the error
message is:

```
Error: failed to inspect container : Unable to read inspect data: json: cannot unmarshal array into Go value of type types.ContainerJSON
```

Totally confusion.

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

tianon committed:
Rename Dockerfile.arm to Dockerfile.armhf to more clearly reflect the specific subversion of ARM it targets

Also, fix up some minor whitespace consistency issues, remove a little cruft, and update GOARM for armhf to 7 so that we're explicit.

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

qq690388648 committed:
replace the os.Stdout with stdout to adapt platform

Signed-off-by: Sun Gengze <690388648@qq.com>

thaJeztah committed:
Merge pull request #18925 from thaJeztah/please-dont-plus-one

Be more explicit about "+1" comments

thaJeztah committed:
add new impact/distribution label

Adds the new label to the documentation. The
impact/distribution label is intended for changes
that affect the image-format or interaction
with the registry (distribution).

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

vdemeester committed:
Merge pull request #18985 from hqhq/hq_handle_dockerCmdInDir

Handle error for dockerCmdInDir

thaJeztah committed:
Merge pull request #18984 from coolljt0725/fix_daemon_start

Fix daemon failed to start with error "layer does not exist"

thaJeztah commented on docs/examples/apt-cacher-ng.md:
```diff
@@ -53,7 +54,8 @@ use:
 
 To get your Debian-based containers to use the proxy, you can do one of
-three things
+four things. Here, `dockerhost` is an IP address or a FQDN of host running docker daemon
```
```
Here, `dockerhost` is the IP address or FQDN of a host running docker daemon
which acts as an APT proxy server.
```

Akasurde replied:
OK

thaJeztah commented on docs/examples/apt-cacher-ng.md:
```diff
@@ -61,6 +63,7 @@ three things
    `http_proxy=http://dockerhost:3142/`
 3. Change your `sources.list` entries to start with
    `http://dockerhost:3142/`
+4. Linking Debian-based containers with Proxy using `--links`
```
perhaps

```
4. Link Debian-based containers to the APT proxy container using `--link`
```

Akasurde replied:
OK

thaJeztah commented on docs/examples/apt-cacher-ng.md:
```diff
@@ -80,6 +83,15 @@ which obey `http_proxy`, such as `curl`, `wget` and others:
+Here, the value provided to the `--link` flag is source_container_name:container_alias_name,
+where test_apt_cacher_ng is source_container_name and apt_proxy is container_alias_name.
+test_apt_cacher_ng was the name of first container that we launched earlier. The container_alias_name
+could be any name of your choice.
```
I think we should try to avoid explaining how link-aliases work.

@moxiegirl may have some suggestions on how to re-word this

Akasurde replied:
OK, I will wait for @moxiegirl comments.

moxiegirl replied:
@Akasurde thanks for the work.  Links are a legacy feature that are likely to be deprecated or changed in the future.  Any problem with suggestion people use a custom network?

thaJeztah commented:
Thanks @Akasurde I left some comments, but @moxiegirl probably will have additional suggestions.

I'm also wondering if we shot add an extra example (or *instead*) using the new networking / container discovery.

calavera committed:
Merge pull request #18952 from coolljt0725/fix_stats_update

Fix docker stats show wrong memory limit when do docker update

Commit:
Merge pull request #18994 from thaJeztah/add-new-impact-labels

add new impact/distribution label

Commit:
Merge pull request #18987 from tianon/armhf

Rename Dockerfile.arm to Dockerfile.armhf to more clearly reflect the specific subversion of ARM it targets

thaJeztah committed:
Merge pull request #18764 from donovan/fix-networkingcontainers-ip-addresses

fix incorrect ip addresses in networkingcontainers

calavera committed:
Merge pull request #18991 from qq690388648/master

replace the os.Stdout with stdout to adapt platform

calavera committed:
Merge pull request #18714 from jecarey/16756-docker_cli_pull_test

Use of checkers in docker_cli_pull_test.go

calavera committed:
Merge pull request #18158 from mauri/add_owner

ADD files to a folder doesn't set correct UID and GID

tonistiigi committed:
Fix missing comment in docker inspect

Fixes #18571

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

thaJeztah committed:
Merge pull request #18964 from hqhq/hq_fix_memory_swap_doc2

Fix docs for memory-swap

calavera committed:
Move test out of the windows build.

Because it can still run on windows server and fail because it doesn't
have `chown`.

Signed-off-by: David Calavera <david.calavera@gmail.com>

Commit:
Merge pull request #19003 from calavera/fix_windows_build

Move test out of the windows build.

aboch committed:
Properly initialize IpamConf structure in initBridgeDriver

Signed-off-by: Alessandro Boch <aboch@docker.com>

calavera committed:
Merge pull request #18999 from tonistiigi/fix-comment-in-inspect

Fix missing comment in docker inspect

jen20 committed:
Fix downstream client API build errors on Solaris

The client API at fsouza/go-dockerclient has dependencies on packages in
the docker/docker repository which currently do not build on Solaris. In
particular, stat_unsupported.go makes use of the Mtimespec field of the
syscall.Stat_t struct, which is not present on Solaris, and a number of
Unix-specific packages do not list Solaris in their compile targets.

This commit adds enough support to be able to build
fsouza/go-dockerclient on SmartOS using Go 1.5.1 without affecting other
platforms.

Signed-off-by: James Nugent <james@jen20.com>

timwangdev committed:
Update LICENSE date

Signed-off-by: Tim <timwangdev@gmail.com>

thaJeztah committed:
Merge pull request #19015 from timwangdev/patch-1

🎉 Happy New Year!  🎉

Commit:
Use constant instead of "scratch"

Move NoBaseImageSpecifier to a common spot and then use it instead of
"scratch" in a couple of places.

Signed-off-by: Doug Davis <dug@us.ibm.com>

cpuguy83 committed:
Merge pull request #18836 from vdemeester/18829-info-authz-plugins

Add authorization plugins to docker info

AvdN committed:
fix typo

Signed-off-by: Anthon van der Neut <a.van.der.neut@ruamel.eu>

Commit:
Merge pull request #19007 from aboch/npe

Properly initialize IpamConf structure in initBridgeDriver

vdemeester committed:
Merge pull request #19006 from jen20/f-solaris-stat

Fix downstream client API build errors on Solaris

runcom committed:
daemon: info: remove exported getter

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

vdemeester committed:
Merge pull request #19019 from duglin/constScratch

Use constant instead of "scratch"

Commit:
Merge pull request #19027 from runcom/remove-getter

daemon: info: remove exported getter

Commit:
Merge pull request #19028 from AvdN/patch-1

fix typo

vdemeester committed:
Merge pull request #18932 from wenchma/same_name_of_container_image

Update integration tests when container and image have same name

thaJeztah committed:
Merge pull request #18891 from HackToday/18872-fix-volume-cmd

Fix the restore data example

albers committed:
Compute plugin list in bash completion for `docker daemon --authz-plugin`

Signed-off-by: Harald Albers <github@albersweb.de>

calavera committed:
Add Ian Murdock to the names generator.

:heart: :cry:

Signed-off-by: David Calavera <david.calavera@gmail.com>

estesp committed:
Merge pull request #19036 from calavera/thank_you_ian

Thanks for everything Ian.

tianon committed:
Merge pull request #19035 from albers/completion-authorization-plugins

Compute plugin list in bash completion for `docker daemon --authz-plugin`

Commit:
Creating a man page for the configuration file

Signed-off-by: Mary Anthony <mary@docker.com>

Adding comments. Fixing bit in daemon

Signed-off-by: Mary Anthony <mary@docker.com>

Arrrgggh

Signed-off-by: Mary Anthony <mary@docker.com>

thaJeztah committed:
Merge pull request #18387 from wenchma/18385-improve_help_msg

Improvement for docker subcommand's help messages

thaJeztah committed:
Merge pull request #19037 from moxiegirl/15666-file-carry

Creating a man page for the configuration file

thaJeztah committed:
Merge pull request #15666 from vdemeester/3519-configurable-escape

Implement configurable escape key for attach/exec

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "i11556" git@github.com:Akasurde/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~98
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

Akasurde committed:
Merge branch 'i11556' of https://github.com/Akasurde/docker into i11556

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde commented:
Closing this PR in favor of #19041
